src/viam/app/app_client.py

```diff
@@ -255,17 +342,33 @@ def __init__(self, channel: Channel, metadata: Mapping[str, str], location_id: O
         self._metadata = metadata
         self._app_client = AppServiceStub(channel)
         self._location_id = location_id
+        self._channel = channel
 
     _app_client: AppServiceStub
     _metadata: Mapping[str, str]
     _location_id: Optional[str]
+    _channel: Channel
+    _organization_id: Optional[str] = None
+
+    def close(self) -> None:
+        self._channel.close()
 
-    async def get_user_id_by_email(self):
+    async def _get_organization_id(self) -> str:
+        return self._organization_id if self._organization_id else await self._request_organization_id()
```

Comment:
I think that's already what is happening..? self._organization_id is initially None when the app_client is constructed. It stays None until the user makes a request that requires the org_id. Once that request is made, we check to see if self.org_id is None or not. If it is NOT none, we just return it, because we already have it "cached". If it IS none, then we have to make this request (it's the only way to get the org_id). This request should only ever happen once, because once its set for the first time, that cached value is always returned.